### ISSUE TITLE
Issue building with polymer cli

### ISSUE THREAD

**Issue by zacktayl** [splitio/javascript-client#229]
Build Environment:
* @polymer/polymer: 3.1.0
* polymer-cli: 1.9.6

Steps to reproduce:
1. `npm i @splitsoftware/splitio@10.6.0`
2. Import via es module: `import { SplitFactory } from '@splitsoftware/splitio';`
3. run `polymer build`
4. The following error appears: 
```Error: ENOENT: no such file or directory, open '/Users/zacktaylor/projects/say3key/frontend/events'```

This seems to be an issue with what Polymer CLI's module resolution system expects - but after debugging this for quite some time, I can't find anything obvious. Any thoughts on what may be the issue?

* Note - I am able to get the [React-Redux-SDK-Example project](https://github.com/splitio/react-redux-sdk-examples) running.

**Comment by NicoZelaya** [comment]
Hey @zacktayl ! 

First of all, thank you for flagging and sorry to hear you're experiencing some issues. I'm able to reproduce and will be investigating to check what's going on. 
Let me know if you need a way to work around it for now, I think you should be able to use the CDN package in the meantime.

I'll get back to you once I have an update! 

Best,
Nico.

**Comment by NicoZelaya** [comment]
Hey @zacktayl ! 

I took a look at what's going on, it seems like the way Polymer is performing the build defers significantly from webpack and other bundlers that can recognize the right "path" for an isomorphic app. 

**TL;DR:** 
Try using the CDN package by adding the script tag to your html before loading your own bundle. 
`<script src="//cdn.split.io/sdk/split-10.6.0.min.js"></script>`

On your code, you can use the global variable `splitio` the exact same way you would use SplitFactory:
```
   const factory = splitio({
      core: {
        authorizationKey: '<Your api key>',
        key: 'polymer_tests'
      }
    });

   const client = factory.client();
   client.on(client.Event.SDK_READY, () => {
      // Get your treatments safely
   });
```
Unless you have a strong motive not to use the CDN package, which is already bundled to work on Browsers and should download pretty fast from the CDN, let us know and we'll try another approach. 


**But what's going on?**
The full explanation would be that Polymer is trying to load the "Node code path" of the SDK, which in turn tries to import the "events" module from NodeJS. 
There are a few shims and even an npm package that replicates that NodeJS module, but it won't work as expected for you even if you install it. 
If you take a look to what's on your `node_modules\@splitsoftware\splitio` folder, you'll see a few package.json files with this format:
```
{
  "main": "./node.js",
  "browser": "./browser.js"
}
```

What we do there is tell NodeJS to just run the "Node version" of that module (the main field), while we tell bundlers to use the "Browser version". This is how, for example, Webpack handles those fields: https://webpack.js.org/configuration/resolve/#resolvemainfields

While researching I found a lot of posts about how to use Webpack for Polymer builds. IF you are using webpack (or migrate to it eventually) and you're still experience the same issue, check those fields. In case that does not do the trick, please let me know so I can work with you towards a resolution.

If you have any more questions or you hit a different problem, you're more than welcome to reach out and we'll find a way to make it work the way you need. Otherwise let us know how it goes!!

Have a wonderful day,
Nico.

**Comment by zacktayl** [comment]
Thanks for the in-depth answer @NicoZelaya! For now I've gone with the CDN approach and that seems to be working great. 

At the moment I do not plan to move to Webpack for this particular app; but if I do, I will make sure to follow the resources you provided!

Thanks again,
Zack
